src/api/app/controllers/webui/monitor_controller.rb

```diff
@@ -77,40 +76,34 @@ def update_building
     render json: workers
   end
 
-  def gethistory(key, range, cache = 1)
-    cachekey = key + "-#{range}"
-    Rails.cache.delete(cachekey, shared: true) unless cache
-    Rails.cache.fetch(cachekey, expires_in: (range.to_i * 3600) / 150, shared: true) do
-      hash = StatusHistory.history_by_key_and_hours(key, range)
-      hash.sort_by { |a| a[0] }
-    end
-  end
-
   def events
     check_ajax
     data = {}
 
-    arch = params[:arch] || ''
+    arch = Architecture.find_by(name: params.fetch(:arch, @default_architecture))
```

Comment:
As @coolo pointed out, the behavior is now different, but I think it makes sense to have a default architecture.